lib/components/transistor.ts

```diff
@@ -0,0 +1,30 @@
+import { voltage, current } from "circuit-json"
+import {
+  type CommonComponentProps,
+  commonComponentProps,
+} from "lib/common/layout"
+import { expectTypesMatch } from "lib/typecheck"
+import { z } from "zod"
+
+export interface TransistorProps extends CommonComponentProps {
+  transistorType: "NPN" | "PNP"
+  pinLabels: Record<"emitter" | "base" | "collector", string>
+  vce?: number | string
+  vbe?: number | string
+  ic?: number | string
```

Comment:
We try not to abbreviate unless the abbreviation is extremely well known

Reply:
done.